src/Routes/Dashboard/SampleTabRoute.js

```diff
@@ -165,9 +144,13 @@ const SampleTabRoute = ({ setMadeChanges, defaults, onChange }) => {
                     ouiaId="enable-cloud-connector"
                     aria-label="Enable Cloud Connector"
                     isChecked={enableCloudConnector}
-                    onChange={() =>
-                      setEnableCloudConnector(!enableCloudConnector)
-                    }
+                    onChange={() => {
+                      const newEnableCloudConnector = !enableCloudConnector;
+                      setEnableCloudConnector(() => newEnableCloudConnector);
```

Comment:
I would use the setState callback value instead of negating current state. Its safer.
```jsx
setEnableCloudConnector((enableCloudConnector) => !enableCloudConnector)
```

And the `setConnectToInsights(() => true)` can be either one within the callback or in `useEffect`. Yes, we will add one render, but it should not cause any performance degradation in this case.